From: Michael Olbrich <m.olbrich@pengutronix.de>
Subject: fix PPC build

From upstream SVN:
------------------------------------------------------------------------
Author: slouken
Date: 2009-12-11 07:14:13 -0800 (Fri, 11 Dec 2009)
New Revision: 5374

Modified:
   branches/SDL-1.2/configure.in
Log:
Fixed bug #895

 Markus Rothe      2009-12-07 09:32:46 PST

Check if libspe2 is available

Building on PowerPC64 results in:

[...]
In file included from ./src/video/ps3/SDL_ps3events_c.h:33,
                 from ./src/video/ps3/SDL_ps3video.c:32:
./src/video/ps3/SDL_ps3video.h:34:21: error: libspe2.h: No such file or
directory
In file included from ./src/video/ps3/SDL_ps3events_c.h:33,
                 from ./src/video/ps3/SDL_ps3video.c:32:
./src/video/ps3/SDL_ps3video.h:75: error: expected specifier-qualifier-list
before 'spe_context_ptr_t'
./src/video/ps3/SDL_ps3video.c:78: error: expected '=', ',', ';', 'asm' or
'__attribute__' before 'fb_writer_spu'
./src/video/ps3/SDL_ps3video.c: In function 'PS3_VideoInit':
./src/video/ps3/SDL_ps3video.c:167: error: 'spu_data_t' has no member named
'program'
./src/video/ps3/SDL_ps3video.c:167: error: 'fb_writer_spu' undeclared (first
use in this function)
[...]

The problem is that the ps3 video driver is being build regardless if libspe2
is installed or not. The attached patch fixes this problem.

[ Note: The other headers that are being checked for are coming from the linux
headers package on powerpc64. As a result these are also available if you're on
a generic powerpc64 machine and not a PS3 only. ]
------------------------------------------------------------------------

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>

---
 configure.in |    1 +
 1 file changed, 1 insertion(+)

Index: b/configure.in
===================================================================
--- a/configure.in
+++ b/configure.in
@@ -1317,6 +1317,7 @@ CheckPS3()
     AC_TRY_COMPILE([
       #include <linux/fb.h>
       #include <asm/ps3fb.h>
+      #include <libspe2.h>
     ],[
     ],[
       video_ps3=yes
